src/main/kotlin/com/bridgecrew/services/CheckovScanService.kt

```diff
@@ -0,0 +1,188 @@
+        if (result.contains("Please check your API token")) {
+            LOG.warn("Please check you API token\n\n" +
+                    " $result")
+            project.messageBus.syncPublisher(CheckovScanListener.SCAN_TOPIC).scanningError()
+            return
+        }
+        if (errorCode != 0 || result.contains("[ERROR]")) {
+            LOG.warn("Error scanning file\n" +
+                    "To report: open an issue at https://github.com/bridgecrewio/checkov-jetbrains-ide/issues\n\n $result")
+            project.messageBus.syncPublisher(CheckovScanListener.SCAN_TOPIC).scanningError()
+            return
+        }
+        try {
+            if (filePath == currentFile) {  // To show only the last run of checkov ( on the opened file)
+                val filePathRelativeToProject = filePath.replace(project.basePath!!, "")
+                val (resultsGroupedByResource, resultsLength) = getGroupedResults(result,
+                    project,
+                    filePathRelativeToProject)
+                project.service<ResultsCacheService>()
+                    .deleteAll() // TODO remove after MVP, where we want to display only one file results
+                project.service<ResultsCacheService>()
+                    .setResult(filePathRelativeToProject, resultsGroupedByResource)
+                project.messageBus.syncPublisher(CheckovScanListener.SCAN_TOPIC).scanningFinished()
+                if (isFirstRun) {
+                    CheckovNotificationBalloon.showError(project, resultsLength)
+                    isFirstRun = false
+                }
+            }
+        } catch (e: JSONException) {
+            LOG.warn("Error parsing checkov results \n" +
+                    "Raw response: $result\n" +
+                    "To report: open a issue at https://github.com/bridgecrewio/checkov-jetbrains-ide/issues\n\n")
+            e.printStackTrace()
+            project.messageBus.syncPublisher(CheckovScanListener.SCAN_TOPIC).scanningError()
+        } catch (e: CheckovResultParsingException){
+            project.messageBus.syncPublisher(CheckovScanListener.SCAN_TOPIC).scanningParsingError()
+        } catch (e: CheckovResultException) {
+            project.messageBus.syncPublisher(CheckovScanListener.SCAN_TOPIC).scanningFinished(filePath)
+        }
+    }
+
+
+    private fun getGroupedResults(res: String, project: Project, relativeFilePath: String): Pair<ResourceToCheckovResultsList, Int> {
+        val listOfCheckovResults = getFailedChecksFromResultString(res)
+        return Pair(groupResultsByResource(listOfCheckovResults, project, relativeFilePath), listOfCheckovResults.size)
+    }
+
+    private fun prepareExecCommand(filePath: String, project: Project, apiToken: String, pluginVersion: String): ArrayList<String> {
+        val execCommand = selectedCheckovRunner!!.getExecCommand(filePath, apiToken, gitRepo, pluginVersion)
```

Comment:
wouldn't that always send the default git repo name?

Reply:
No, as in startup we run getRepo and update this value in the gitUtils
`project.service<CheckovScanService>().gitRepo = result`